src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+
+        this.state = {
+            steps: [StepType.IOUAmount, StepType.IOUParticipants, StepType.IOUConfirm],
+            currentStepIndex: 0,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+
+        this.getTitleForStep = this.getTitleForStep.bind(this);
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Returns the title for the currently selected page
+     *
+     * @return {String}
+     */
+    getTitleForStep() {
+        switch (this.state.steps[this.state.currentStepIndex]) {
+            case StepType.IOUAmount:
+                return 'Amount';
+            case StepType.IOUParticipants:
+                return 'Participants';
+            case StepType.IOUConfirm:
+                return 'Confirm';
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Navigate to the next IOU step
+     */
+    navigateToPreviousStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex - 1}));
+    }
+
+    /**
+     * Navigate to the previous IOU step
+     */
+    navigateToNextStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex + 1}));
+    }
+
+    render() {
+        return (
+            <>
+                <View style={[styles.headerBar, true && styles.borderBottom]}>
+                    <View style={[
+                        styles.dFlex,
+                        styles.flexRow,
+                        styles.alignItemsCenter,
+                        styles.flexGrow1,
+                        styles.justifyContentBetween,
+                        styles.overflowHidden,
+                    ]}
+                    >
+                        {this.state.currentStepIndex > 0
+                        && (
+                            <TouchableOpacity
+                                onPress={() => this.navigateToPreviousStep()}
+                                style={[styles.touchableButtonImage]}
+                            >
+                                <Icon src={BackArrow} />
+                            </TouchableOpacity>
+                        )}
+                        <Header title={this.getTitleForStep()} />
+                        <View style={[styles.reportOptions, styles.flexRow]}>
+                            <TouchableOpacity
+                                onPress={redirectToLastReport}
+                                style={[styles.touchableButtonImage]}
+                            >
+                                <Icon src={Close} />
+                            </TouchableOpacity>
+                        </View>
+                    </View>
+                </View>
+                {this.state.steps[this.state.currentStepIndex] === StepType.IOUAmount
```

Comment:
I'm not too big of a fan of this, and I think it's mostly because the render method of this component is getting very big. My biggest suggestion to clean this up a bit is to utilize the router more. @marcaaron might have a different opinion here, especially since we are changing to `react-navigation`, but I'll throw out this suggestion anyway.

1. Have routes added for `iou/split/amount` `iou/split/participants` `iou/split/confirm`
2. Use a `<Switch>` here to display the proper component based on the route matching
3. When navigating steps, use `App.redirect()` to go to the next/prev step

This will give you a few extra benefits:
- Being able to navigate the form just like navigating through the app (browser back button, gestures, etc.)
- Being able to refresh the page and retain the step (maybe good, maybe bad, but with Onyx it would be cool)

Reply:
Totally agree with your concerns and the mentioned benefits. 

This Component was actually planned with react-navigation in mind from the beginning, we'll take advantage of the above mentioned benefits as soon as it [react-navigation] gets merged. An additional benifit is that we'll get native page-to-page animations for free too! (using [tab-navigation](https://reactnavigation.org/docs/tab-based-navigation/))

Reply:
Resolving this thread, as discussion has moved to [this comment](https://github.com/Expensify/Expensify.cash/pull/1561#discussion_r588705127).

Reply:
Resolving this thread, as discussion has moved to [this comment](https://github.com/Expensify/Expensify.cash/pull/1561#discussion_r588705127).